src/coreComponents/linearAlgebra/solvers/PreconditionerJacobi.hpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2019 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2019 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2019 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All right reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#ifndef GEOSX_LINEARALGEBRA_SOLVERS_PRECONDITIONERJACOBI_HPP_
+#define GEOSX_LINEARALGEBRA_SOLVERS_PRECONDITIONERJACOBI_HPP_
+
+#include "linearAlgebra/interfaces/LinearOperator.hpp"
+#include "linearAlgebra/solvers/PreconditionerBase.hpp"
+
+namespace geosx
+{
+
+/**
+ * @brief Common interface for identity preconditioning operator
+ * @tparam LAI linear algebra interface providing vectors, matrices and solvers
+ */
+template< typename LAI >
+class PreconditionerJacobi : public PreconditionerBase< LAI >
```

Comment:
Nothing particular, except is currently available only in hypre. Being trivial (on CPU), I coded it.

Reply:
As far as I can tell, we have access to Jacobi implemented in all 3 interfaces, although I haven't personally ensured they're all working properly. In any case, we can keep this for now since it's working well for you. But running linear algebra entirely on GPU should always be in the crosshairs.

Reply:
Ok. I'll check the LAI versions of `jacobi`, maybe I did something wrong in the past (when I decided to implement _my_ Jacobi).

Reply:
Now I'm using `Jacobi` from LAI. `PreconditionerJacobi.hpp` is still there ... we can remove it if required.